packages/waku/tests/vite-plugin-rsc-transform-internals.test.ts

```diff
@@ -68,10 +66,117 @@ export const log = (mesg) => {
       `);
   });
 
+  test('server action in object', async () => {
+    const code = `
+import type { ReactNode } from 'react';
+import { createAI } from 'ai/rsc';
+
+const AI = createAI({
+  actions: {
+    foo: async () => {
+      'use server';
+      return 0;
+    },
+  },
+});
+
+export function ServerProvider({ children }: { children: ReactNode }) {
+  return <AI>{children}</AI>;
+}
+`;
+    expect(await transform(code, '/src/App.tsx', { ssr: true }))
+      .toMatchInlineSnapshot(`
+        "import type { ReactNode } from 'react';
+        import { createAI } from 'ai/rsc';
+        import { registerServerReference as __waku_registerServerReference } from 'react-server-dom-webpack/server.edge';
+        export const __waku_action1 = __waku_registerServerReference(async ()=>{
+            return 0;
+        }, "/src/App.tsx", "__waku_action1");
+        const AI = createAI({
+            actions: {
+                foo: __waku_action1.bind(null)
+            }
+        });
+        export function ServerProvider({ children }: {
+            children: ReactNode;
+        }) {
+            return <AI>{children}</AI>;
+        }
+
+        if (typeof ServerProvider === 'function') {
+          __waku_registerServerReference(ServerProvider, '/src/App.tsx', 'ServerProvider');
+        }
+        "
+      `);
+  });
+
+  test('create server with bind', async () => {
+    const code = `
+'use server';
+import { InternalProvider } from './shared.js';
+import { jsx } from 'react/jsx-runtime';
+
+async function innerAction({ action }, ...args) {
+  'use server';
```

Comment:
Okay, you want to support this double `use server` case. I thought about it, but I was not sure if it's valid.

Reply:
It's valid in Next.js I think🤔

Reply:
I wrote down all cases in sheet
<img width="850" alt="image" src="https://github.com/dai-shi/waku/assets/14026360/e09af0d3-fd88-4cd7-8aea-765b89d52c93">

Reply:
yeah, it's reasonable with the object param case.